app/src/main/java/pl/szczodrzynski/edziennik/ui/dialogs/timetable/GenerateBlockTimetableDialog.kt

```diff
@@ -373,25 +376,31 @@ class GenerateBlockTimetableDialog(
 
             val today = Date.getToday().stringY_m_d
             val now = Time.getNow().stringH_M_S
-
-            val outputDir = Environment.getExternalStoragePublicDirectory("Szkolny.eu").apply { mkdirs() }
-            val outputFile = File(outputDir, "plan_lekcji_${app.profile.name}_${today}_${now}.png")
+            val filename = "plan_lekcji_${app.profile.name}_${today}_${now}.png"
+            val resolver: ContentResolver = activity.applicationContext.contentResolver
+            val values = ContentValues()
+            values.put(MediaStore.MediaColumns.MIME_TYPE, "image/png")
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                values.put(MediaStore.MediaColumns.DISPLAY_NAME, filename)
+                values.put(MediaStore.MediaColumns.RELATIVE_PATH, Environment.DIRECTORY_PICTURES + File.separator + "Szkolny.eu")
+            } else {
+                val picturesDirectory = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath + File.separator + "Szkolny.eu"
+                File(picturesDirectory).mkdirs()
+                values.put(MediaStore.MediaColumns.DATA, picturesDirectory + File.separator + filename)
+            }
 
             try {
-                val fos = FileOutputStream(outputFile)
+                val uri = resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values)!!
+                val fos = resolver.openOutputStream(uri)
```

Comment:
Isn't there a KTX for streams? Something like `resolver.openOutputStream(uri).use { }`.
Also, you're using `!!` too much.

Reply:
Something like this?

Reply:
I think the idea behind `.use {}` is that you don't have to close it manually.

Reply:
Okay, I never used it before.